### DISCUSSION TITLE
Tracking loading progress

### DISCUSSION THREAD

**Discussion by sseerrppeenntt** [uwdata/mosaic#530] [Q&A]
Hello, I just discovered Mosaic and it looks very promising, as I am interested in the topic of large-volume data visualization on the web ❤.

When going through the examples I noticed that the fetched data is split into multiple XHR requests. In the browser, I know how to track the progress of one XHR with the `progress` event, but I am not sure how does one determine how many requests will there be in total.

UX-wise it would be very beneficial to show the loading progress to the user - to be calmed that the application didn't just lag out and is doing work in the background (especially when handling large datasets).

Is there a way to track the total progress of the fetched data; to determine how many requests will there be in total?

**Comment by domoritz** [comment]
The multiple requests are coming from DuckDB if you use the WASM connector. We don't have a general progress API right now. I think we should start by showing a spinner when there is a request that takes more than a few ms an eventually a progress bar but that's fairly hard in general since queries can take an unpredictable amount of time.
